cmd/image-builder/repos.go

```diff
@@ -39,5 +75,25 @@ var newRepoRegistry = func(dataDir string) (*reporegistry.RepoRegistry, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// XXX: this should probably go into manifestgen.Options as
+	// a new Options.ExtraRepoConf eventually (just like OverrideRepos)
+	for _, repo := range extraRepos {
+		// XXX: this loads the extra repo unconditionally to all
+		// distro/arch versions. we do not know in advance where
+		// it belongs to
+		extraRepo, err := parseExtraRepo(repo)
+		if err != nil {
+			return nil, err
+		}
+		for _, repoArchConfigs := range conf {
+			for arch := range repoArchConfigs {
+				archCfg := repoArchConfigs[arch]
+				archCfg = append(archCfg, extraRepo...)
+				repoArchConfigs[arch] = archCfg
+			}
+		}
```

Comment:
I don't think that the extra repos should be mixed with the repo configs loaded from the FS. This will make them being used for the 1st transaction depsolving and would effectively extend the "system" repos. These extra repos should be treated separately and should be tagged by default with the payload pipeline names specific to a particular image type, when generating the manifest.

This specifically relates to my comment https://github.com/osbuild/images/pull/1153#discussion_r1926947542

Reply:
I think this got resolved in the most recent discussion that we would allow to add base repos as this is already possible via the "--data-dir" and /etc/image-builder/repositories flags.